core/src/main/java/ai/timefold/solver/core/impl/localsearch/decider/acceptor/restart/UnimprovedMoveCountRestartStrategy.java

```diff
@@ -0,0 +1,78 @@
+package ai.timefold.solver.core.impl.localsearch.decider.acceptor.restart;
+
+import java.time.Clock;
+
+import ai.timefold.solver.core.api.score.Score;
+import ai.timefold.solver.core.impl.localsearch.scope.LocalSearchMoveScope;
+import ai.timefold.solver.core.impl.localsearch.scope.LocalSearchPhaseScope;
+import ai.timefold.solver.core.impl.localsearch.scope.LocalSearchStepScope;
+import ai.timefold.solver.core.impl.solver.scope.SolverScope;
+
+/**
+ * Restart strategy, which exponentially increases the move count values that trigger the restart process.
+ */
+public class UnimprovedMoveCountRestartStrategy<Solution_> extends AbstractGeometricRestartStrategy<Solution_> {
+
+    protected long lastImprovementMoveCount;
+    private Score<?> currentBestScore;
+
+    public UnimprovedMoveCountRestartStrategy() {
+        this(Clock.systemUTC());
+    }
+
+    protected UnimprovedMoveCountRestartStrategy(Clock clock) {
+        // 50k moves as the multiplier
+        super(clock, 50_000);
+    }
+
+    @Override
+    public void phaseStarted(LocalSearchPhaseScope<Solution_> phaseScope) {
+        super.phaseStarted(phaseScope);
+        currentBestScore = phaseScope.getBestScore();
+    }
+
+    @Override
+    public void stepStarted(LocalSearchStepScope<Solution_> stepScope) {
+        this.currentBestScore = stepScope.getPhaseScope().getBestScore();
+    }
+
+    @Override
+    @SuppressWarnings({ "rawtypes", "unchecked" })
+    public void stepEnded(LocalSearchStepScope<Solution_> stepScope) {
+        // Do not update it during the grace period
+        if (isGracePeriodFinished() && ((Score) stepScope.getScore()).compareTo(currentBestScore) > 0) {
+            lastImprovementMoveCount = stepScope.getPhaseScope().getSolverScope().getMoveEvaluationCount();
+        }
+    }
+
+    @Override
+    public void solvingStarted(SolverScope<Solution_> solverScope) {
+        super.solvingStarted(solverScope);
+        lastImprovementMoveCount = 0;
+    }
+
+    @Override
+    public boolean process(LocalSearchMoveScope<Solution_> moveScope) {
+        var currentMoveCount = moveScope.getStepScope().getPhaseScope().getSolverScope().getMoveEvaluationCount();
+        if (lastImprovementMoveCount == 0) {
+            lastImprovementMoveCount = currentMoveCount;
+            return false;
+        }
+        if (currentMoveCount - lastImprovementMoveCount >= nextRestart) {
+            logger.debug("Restart triggered with geometric factor {}, scaling factor of {}, move count ({})",
```

Comment:
Let's make this `trace`.

The solver's logging is designed poorly - it uses various solving levels to map to how "deep" you are in the solving process. INFO is phases, DEBUG is steps, TRACE is moves. This is unfortunate, but we can not really change this.

Let's put these things on TRACE, which is the lowest possible level - if people enable this one, they will get everything.